### PR TITLE
ufs-tool: unipro: add Unipro layers attributes read and write

### PR DESCRIPTION
**Pull request by beanhuo**
hi, Avri and Arthur
Based on your comments for the last version, 
1. fixed lots of coding style issues
2. move unipro command parameters checkup to option.c.
3. add more comments

### CODE REVIEW

File: unipro.c [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+	{"DME_TXRX_PA_INIT_REQUEST", 0x5121, (GETTABLE | SETTABLE)},
+	{"DME_TXRX_QoS_COUNT", 0x5122, (GETTABLE | SETTABLE)},
+	{"DME_TXRX_DL_LM_ERROR", 0x5123, (GETTABLE | SETTABLE)},
+	{"DME_QoS_ENABLE", 0x5130, (GETTABLE | SETTABLE)},
+	{"DME_QoS_STATUS", 0x5131, (GETTABLE | SETTABLE)},
+};
+
+/* M-TX/M-RX Capability Attributes */
+static struct ufs_uic_attr_fields mipi_mphy_attrs[] = {
+	// M-PHY TX Capability Attributes
```
Comment (arthur-simchaev-sndk) [inline unipro.c]: Line 108 Please change to /* comment

File: unipro.c [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+	{"TX_Min_SAVE_Config_Time_Capability", 0x000B, (GETTABLE)},
+	{"TX_REF_CLOCK_SHARED_Capability", 0x000C, (GETTABLE)},
+	{"TX_PHY_MajorMinor_Release_Capability", 0x000D, (GETTABLE)},
+	{"TX_PHY_Editorial_Release_Capability", 0x000E, (GETTABLE)},
+	{"TX_Hibern8Time_Capability", 0x000F, (GETTABLE)},
+	{"TX_Advanced_Granularity_Capability", 0x0010, (GETTABLE)},
+	{"TX_Advanced_Hibern8Time_Capability", 0x0011, (GETTABLE)},
+	{"TX_HS_Equalizer_Setting_Capability", 0x0012, (GETTABLE)},
+
+	// M-PHY TX Configuration Attributes
```
Comment (arthur-simchaev-sndk) [inline unipro.c]: Line 129 , please change to /*

File: unipro.c [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+		goto out;
+	}
+
+	memcpy(&uic_rsq, &bsg_rsp.upiu_rsp.uc, UIC_CMD_SIZE);
+	res_code = uic_rsq.argument2 & MASK_UIC_COMMAND_RESULT;
+
+	if (res_code) {
+		__u8 max_code = sizeof(resultcode) /
+			sizeof(struct uic_cmd_result_code);
+
```
Comment (arthur-simchaev-sndk) [inline unipro.c:352]: Maybe you can use ARRAY_SIZE define

**Review by arthur-simchaev-sndk — COMMENTED** [review]
Except of a 3 minor problems it's Ok for me

**Comment by beanhuo** [comment]
hi, Arthur
thanks so much! 
would you help change this for me?  Otherwise, pull request again.

**Comment by arthur-simchaev-sndk** [comment]
No problem , I can change it. But I think, you don't need to do the new PR . Just fix it , amend the commit and push force the commit  .

**Review by arthur-simchaev-sndk — DISMISSED** [review]

File: unipro.c (hunk 3 above)
Comment (beanhuo) [reply unipro.c:352]: hi, Arthur
I saw this macro you defined in ufs_cmds.c.
#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))

If I change this, I should move this macro to .h. then I can change, or I define it in unipro.c again.
that will be redundant.  I want firstly keep my current code, let me fix it later if you are ok. 

thanks for your suggestion, let me see how other reviewers think about.

**Pull request closed without merging**